neuraxle/hyperparams/space.py

```diff
@@ -226,6 +227,61 @@ class HyperparameterSamples(RecursiveDict):
     def __init__(self, *args, separator=None, **kwds):
         super().__init__(*args, separator=separator, **kwds)
 
+    def compress(self, trim_parents=False):
+        """Compresses the HyperparameterSamples representation """
+        grouped_result = self._group_hps_by_step(flat_steps_hps=self.items())
+        return self._convert_to_compressed_format(group_by_step=grouped_result, trim_parents=trim_parents)
+
+    def _group_hps_by_step(self, flat_steps_hps: list) -> dict:
+        """
+        Groups hyperparams
+        :param flat_steps_hps: Flat list of steps and hyper parameter
+        :return: OrderedDict containning grouped hyperparams as values and keys as pipeline stages & steps
+
+        >>>hyper_params = [('AddFeatures__PCA__copy', True), ('AddFeatures__PCA__iterated_power', 'auto')]
+        >>>self._group_hps_by_step(hyper_params = hyper_params)
+        ...OrderedDict([('AddFeatures__PCA', OrderedDict([('copy', True), ('iterated_power', 'auto')]))])
```

Comment:
The OrderedDict (or a list) seems to be a good idea to maintain the order of *steps*. However, for the order of *parameters belonging to the same step*, then I don't think the order is important, and removing the order may make things more readable. 
```suggestion
        ...OrderedDict([('AddFeatures__PCA', dict([('copy', True), ('iterated_power', 'auto')]))])
```